src/App.js

```diff
@@ -1,75 +1,142 @@
 import React from 'react';
+import NewTodo from './NewTodo';
+import TodoList from './TodoList';
+import Filter from './TodosFilter';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todoList: [], allCompleted: false, filter: 'All',
+  };
+
+  handleAddingTodo = (newTodo) => {
+    this
+      .setState(({ todoList }) => ({ todoList: [...todoList, newTodo] }));
+  };
+
+  handleCheckItem = (id) => {
+    this
+      .setState(({ todoList }) => (
+        { todoList: todoList.map(item => (
+          item.id === id ? {
+            ...item, completed: !item.completed,
+          } : item)) }));
+
+    this.setState(({ todoList }) => (
+      { allCompleted: todoList.every(item => item.completed) }
+    ));
+  };
+
+  handleDestroyItem = (id) => {
+    this.setState(({ todoList }) => (
+      { todoList: todoList
+        .filter(item => item.id !== id) }
+    ));
+  };
+
+  handleCompleteAll = () => {
+    this
+      .setState(({ todoList, allCompleted }) => ({
+        allCompleted: !allCompleted,
+        todoList: todoList
+          .map(item => ({
+            ...item, completed: !allCompleted,
+          })),
+      }));
+  };
+
+  handleSelectActive = () => {
```

Comment:
```suggestion
  setFilter = (filter) => {
```
обробляй всі випадки в одному методі, не треба купу методів які роблять одне і те ж